### PR TITLE
Added new exec to read sensor details

### PR DESCRIPTION
**Pull request by carlos-m159**
Output looks like this:
```
{
    "configuration_id": "N/A",
    "firmware_version": "02.XX.XX",
    "model_number": "UAM-05LP",
    "safety_areas_crc": "XXXXXXXX",
    "serial_number": "XXXXXX"
}
```

### CODE REVIEW

File: src/uam/io/tcp_client.cpp [hunk 1]
```diff
@@ -130,18 +130,34 @@ bool TcpClient::connect(const std::string& remote_ip, uint16_t remote_port)
...
   // Open the socket
-  boost::system::error_code error_code;
-  remote_endpoint_.address().to_string();
-  socket_.connect(remote_endpoint_, error_code);
-
-  if (error_code)
   {
-    ROS_ERROR_STREAM("Failed to connect: " << error_code.message());
-    return false;
+    auto connect_future = socket_.async_connect(remote_endpoint_, boost::asio::use_future);
```
Comment (carlos-m159) [inline src/uam/io/tcp_client.cpp:134]: Had to change to async connect to prevent connect from blocking indefinitely.

File: CMakeLists.txt [hunk 2]
```diff
@@ -46,6 +47,11 @@ add_library(uam_workers
   src/uam/io/tcp_client.cpp
   )
 target_link_libraries(uam_workers ${catkin_LIBRARIES})
+
+add_library(uam_driver
```
Comment (svwilliams) [inline CMakeLists.txt:51]: Do you want to export this library in the `catkin_package()` call as well?

File: src/uam/uam_get_safety_configuration_id.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  // Use Boost argparser to read commandline args
+  auto options = po::options_description("Allowed arguments");
+  // clang-format off
+  options.add_options()
+      ("help,h",
+          "produce help message")
+      ("lidar-ip", po::value<std::string>()->default_value("192.168.158.98"),
+          " The IP Address of the lidar server")
+      ("lidar-port", po::value<unsigned int>()->default_value(6543),
+          "The tcp client port number the client/robot is using to communicate with laser");
```
Comment (svwilliams) [inline src/uam/uam_get_safety_configuration_id.cpp]: I think this is actually the port number that the lidar is listening to. Should the default be 10940?

File: src/uam/uam_get_safety_configuration_id.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+{
+  namespace po = boost::program_options;
+
+  // Use Boost argparser to read commandline args
+  auto options = po::options_description("Allowed arguments");
+  // clang-format off
+  options.add_options()
+      ("help,h",
+          "produce help message")
+      ("lidar-ip", po::value<std::string>()->default_value("192.168.158.98"),
```
Comment (svwilliams) [inline src/uam/uam_get_safety_configuration_id.cpp]: This is the strange Parakeet lidar default value. Do you know the default values for a Hokuyo lidar? 192.168.0.10 maybe?

File: src/uam/uam_get_safety_configuration_id.cpp [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        area_number,
+        0,
+        uam::protocol::c_nr_ranges);
+      if (yr_area.has_value())
+      {
+        crcs.append(std::to_string(yr_area->footer.crc));
+      }
+    }
+  }
+  // We got the complete list of areas, swap
```
Comment (svwilliams) [inline src/uam/uam_get_safety_configuration_id.cpp]:
```suggestion
  // We got the complete list of areas
```

File: src/uam/uam_get_safety_configuration_id.cpp [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+         area_type++)
+    {
+      auto yr_area = lidar.getSafetyArea(
+        static_cast<uam::protocol::EYRAreaType>(area_type),
+        area_number,
+        0,
+        uam::protocol::c_nr_ranges);
+      if (yr_area.has_value())
+      {
+        crcs.append(std::to_string(yr_area->footer.crc));
```
Comment (svwilliams) [inline src/uam/uam_get_safety_configuration_id.cpp]: Do I even want to know how long this string is?

**Review by svwilliams — APPROVED** [review]
Looks correct to me

File: src/uam/uam_get_safety_configuration_id.cpp [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  namespace po = boost::program_options;
+
+  // Use Boost argparser to read commandline args
+  auto options = po::options_description("Allowed arguments");
+  // clang-format off
+  options.add_options()
+      ("help,h",
+          "produce help message")
+      ("lidar-ip", po::value<std::string>()->required(),
+          " The IP Address of the lidar server in the form of XXX.XXX.XXX.XX")
```
Comment (aantunes-locus) [inline src/uam/uam_get_safety_configuration_id.cpp]:
```suggestion
          "The IP Address of the lidar server in the form of XXX.XXX.XXX.XXX")
```

**Review by aantunes-locus — APPROVED** [review]

**Pull request merged**